src/content/app-architecture/case-study/ui-layer.md

```diff
@@ -0,0 +1,677 @@
+alt="A diagram showing a View's relationship to a ViewModel.">
+
+Continuing the Home feature example, 
+the following code shows the definition of the ‘HomeScreen' view.
+
+```dart
+// app/lib/ui/home/widgets/home_screen.dart
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({super.key, required this.viewModel});
+
+  final HomeViewModel viewModel;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      //...
+    );
+  }
+}
+```
+
+Most of the time, a view's only inputs should be a `key`, 
+which all Flutter widgets take as an optional argument, 
+and the view's corresponding ViewModel.
+
+### Display UI data in a View
+
+A view depends on a ViewModel for its state. In the Compass app, 
+the ViewModel is passed in as an argument in the view's constructor. 
+The following example code snippet is from the `HomeScreen` widget.
+
+```dart
+// app/lib/ui/home/widgets/home_screen.dart
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({super.key, [!required this.viewModel!]});
+
+  [!final HomeViewModel viewModel;!]
+
+  @override
+  Widget build(BuildContext context) {
+    // ...
+  }
+}
+```
+
+Within the widget, you can access the passed-in bookings from the `viewModel`. 
+In the following code, 
+the `booking` property is being provided to a sub-widget.
+
+```dart
+// app/lib/ui/home/widgets/home_screen.dart
+ 
+@override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      // Some code removed for brevity
+      body: SafeArea(
+        child: ListenableBuilder(
+          listenable: viewModel,
+          builder: (context, _) {
+            return CustomScrollView(
+              slivers: [
+                SliverToBoxAdapter(...),
+                SliverList.builder(
+                   itemCount: [!viewModel.bookings.length!],
+                    itemBuilder: (_, index) => _Booking(
+                      key: ValueKey([!viewModel.bookings[index].id!]),
+                      booking:viewModel.bookings[index],
+                      onTap: () => context.push(Routes.bookingWithId(
+                         viewModel.bookings[index].id)),
+                      onDismissed: (_) => viewModel.deleteBooking.execute(
+                           viewModel.bookings[index].id,
+                         ),
+                    ),
+                ),
+              ],
+            );
+          },
+        ),
+      ),
+```
+
+### Update the UI 
+
+The `HomeScreen` widget listens for updates from the ViewModel with
+the [`ListenableBuilder`][] widget. 
+Everything in the widget subtree under the 
+`ListenableBuilder` widget re-renders when the
+provided [`Listenable`][] changes. In this case, 
+the provided `Listenable` is the ViewModel. 
+Recall that the ViewModel is of type [`ChangeNotifier`][] 
+which is a subtype of the `Listenable` type.
+
+```dart
+// app/lib/ui/home/widgets/home_screen.dart
+ 
+@override
+  Widget build(BuildContext context) {
```

Comment:
The indentation looks off here. Feel free to left align the method, even if it would be indented in the actual file.